x-pack/plugins/observability_onboarding/public/context/create_wizard_context.tsx

```diff
@@ -32,115 +35,104 @@ export interface WizardContext<T, StepKey extends string> {
   };
 }
 
-export function createWizardContext<
-  T,
-  StepKey extends string,
-  InitialStepKey extends StepKey
->({
+export function createWizardContext<T, StepKey extends string>({
   initialState,
-  initialStep,
   steps,
 }: {
   initialState: T;
-  initialStep: InitialStepKey;
   steps: Record<StepKey, ComponentType>;
 }) {
   const context = createContext<WizardContext<T, StepKey>>({
-    CurrentStep: () => null,
+    setCurrentStep: () => {},
     goToStep: () => {},
     goBack: () => {},
     getState: () => initialState,
     setState: () => {},
     getPath: () => [],
-    getUsage: () => ({ timeSinceStart: 0, navEvents: [] }),
+    getUsage: () => ({
+      timeSinceStart: 0,
+      navEvents: new Array<NavEvent<StepKey>>(),
+    }),
   });
 
   function Provider({
     children,
     onChangeStep,
     transitionDuration,
+    basePath,
   }: {
-    children?: ReactNode;
+    children: ReactNode;
     onChangeStep?: (stepChangeEvent: {
       direction: 'back' | 'next';
       stepKey: StepKey;
       StepComponent: ComponentType;
     }) => void;
     transitionDuration?: number;
+    basePath: string;
   }) {
-    const [step, setStep] = useState<StepKey>(initialStep);
-    const pathRef = useRef<StepKey[]>([initialStep]);
+    const history = useHistory();
+    const [step, setStep] = useState<StepKey>();
+    const pathRef = useRef<StepKey[]>([]);
     const usageRef = useRef<ReturnType<WizardContext<T, StepKey>['getUsage']>>({
       timeSinceStart: 0,
-      navEvents: [
-        { type: 'initial', step, timestamp: Date.now(), duration: 0 },
-      ],
+      navEvents: new Array<NavEvent<StepKey>>(),
```

Comment:
is the reason to use the Array constructor for types? you could continue to use the array literal `[] as Array<NavEvent<StepKey>>`. Also if the type is defined here, why does it need to be asserted on line 101?

Reply:
There we are not defining types but initialising the value of `navEvents`.

Reply:
no functional difference, just style